packages/manager/apps/web/client/app/hosting/cdn/flush/hosting-cdn-flush.controller.js

```diff
@@ -17,8 +18,9 @@ export default class HostingFlushCdnCtrl {
   }
 
   flushCdn() {
-    this.Hosting.flushCdn(this.$stateParams.productId)
-      .then(() => {
+    const isV1CDN = get(this.$scope.cdnProperties, 'version', '') === '2013v1';
```

Comment:
Don't we have any other way to pass the `cdnProperty` than `$scope` ? Maybe through a resolve ? 
Also Maybe we can consider creating a `CDN` class to expose this kind of utility

Reply:
i take a look into, the `hosting-cdn-flush.controller.js` had no routing file to use a resolve.
i think they use $scope because it bind to modal controller.